[Lowering] Fix the edge case of empty subgraph split due to dataclass node

yuhuishi-convect opened the pull request:
Summary: Fix the edge case by allowing `call_function` nodes with no deps as graph entry (starter_nodes) in the splitter.

Test Plan:
The test shall pass in the current diff (after fix), and fail in the parent diff (before fix)

```
buck test mode/opt //glow/fb/fx/lowering:split_tests -- test_dataclass_as_graph_entry
```

Rollback Plan:

Differential Revision: D81232435




cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

ezyang reviewed (changes requested):
Needs test

yuhuishi-convect commented:
> Needs test

there are some internal test suites that cover this file. [D81231937](https://www.internalfb.com/diff/D81231937)
but we are lacking tests in the public repo. 
@ezyang would you recommend porting some internal tests here?

ezyang reviewed (approved):
Review automatically exported from Phabricator review in Meta.

yuhuishi-convect committed:
[Lowering] Fix the edge case of empty subgraph split due to dataclass node (#161716)

Summary:

Fix the edge case by allowing `call_function` nodes with no deps as graph entry (starter_nodes) in the splitter.

Test Plan:
The test shall pass in the current diff (after fix), and fail in the parent diff (before fix)

```
buck test mode/opt //glow/fb/fx/lowering:split_tests -- test_dataclass_as_graph_entry
```

```
 buck test //caffe2/test:fx  -- -r test_fx_split
```
After fix:

https://www.internalfb.com/intern/testinfra/testrun/13229324014780441

Before fix:

https://www.internalfb.com/intern/testinfra/testrun/5348024864258769

Rollback Plan:

Reviewed By: ezyang

Differential Revision: D81232435

ezyang commented:
@pytorchbot merge -r

davidberard98 commented:
@pytorchbot merge
